# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/rene/emacs/ppc-oddity.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

To be able to build and run Emacs on PowerPC we need to wipe some strange
outdated ... assumptions ...

  - Rene Rebe <rene@rocklinux.org>

--- emacs-21.3/src/m/macppc.h	2001-10-23 08:43:33.000000000 +0200
+++ emacs-21.3/src/m/macppc.h	2003-09-26 22:26:59.000000000 +0200
@@ -90,26 +90,9 @@
 #define HAVE_TEXT_START
 #endif
 
-/* NAKAJI Hiroyuki <nakaji@tutrp.tut.ac.jp> says this is needed
-   For MkLinux/LinuxPPC.  */
 
-#ifdef LINUX
-#define LINKER $(CC) -nostdlib
-#define LD_SWITCH_MACHINE -Xlinker -m -Xlinker elf32ppc
-/* s/gnu-linux.h defines this to `-z nocombreloc' which does not work here
-   because prefix-args is not used.  */
-#undef LD_SWITCH_SYSTEM_TEMACS
-#define LD_SWITCH_MACHINE_TEMACS -Xlinker -znocombreloc
-#endif
-
-#if 0  /* This breaks things on PPC GNU/Linux ecept for Yellowdog,
-	  even with identical GCC, as, ld.  Let's take it out until we
-	  know what's really going on here.  */
-/* GCC 2.95 and newer on GNU/Linux PPC changed the load address to
-   0x10000000.  */
 #if defined __linux__
 #if __GNUC__ > 2 || (__GNUC__ == 2 && __GNUC_MINOR__ >= 95)
 #define DATA_SEG_BITS  0x10000000
 #endif
-#endif
 #endif /* 0 */
